pkg/operator/controller/externaldns/deployment.go

```diff
@@ -65,7 +69,12 @@ var sourceStringTable = map[operatorv1alpha1.ExternalDNSSourceType]string{
 func (r *reconciler) ensureExternalDNSDeployment(ctx context.Context, namespace, image string, serviceAccount *corev1.ServiceAccount, externalDNS *operatorv1alpha1.ExternalDNS) (bool, *appsv1.Deployment, error) {
 	nsName := types.NamespacedName{Namespace: namespace, Name: controller.ExternalDNSResourceName(externalDNS)}
 
-	desired, err := desiredExternalDNSDeployment(namespace, image, serviceAccount, externalDNS)
+	secret, err := r.extractCredentialsSecret(ctx, externalDNS)
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to extract credentials secret: %w", err)
+	}
+
+	desired, err := desiredExternalDNSDeployment(namespace, image, serviceAccount, secret, externalDNS)
```

Comment:
`desiredExternalDNSDeployment` just needs the secret's name so that it can put the name in the volume and env specs, right?  Is there a need to verify that the secret exists and pass it to `desiredExternalDNSDeployment` instead of just passing in the name of the secret?

Reply:
The secret is supposed to be brought by the other controller: `credentials-secret`. In the worth case scenario (secret provider in CR doesn't exist) we end up with a POD which cannot start because of a missing secret. I think it's more visible (the the creator of CR) than not creating a deployment if the secret is not there.